test: randomly failed 'testTypeConvertSuite.TestConvertJSONToDecimal' in check_dev2

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.


```
[2020-02-03T07:42:11.123Z] FAIL: convert_test.go:973: testTypeConvertSuite.TestConvertJSONToDecimal

[2020-02-03T07:42:11.123Z] 

[2020-02-03T07:42:11.123Z] convert_test.go:990:

[2020-02-03T07:42:11.123Z]     c.Assert(casted.Compare(tt.Out), Equals, 0)

[2020-02-03T07:42:11.123Z] ... obtained int = -1

[2020-02-03T07:42:11.123Z] ... expected int = 0
```
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/22807/pipeline/
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/22797/pipeline
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/22815/pipeline
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/23021/pipeline


2. What did you expect to see?



3. What did you see instead?



4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

Another appearance on my laptop.

```
[2020/02/10 21:37:30.236 +08:00] [ERROR] [terror.go:363] ["encountered error"] [error="[types:8029]Bad Number"] [errorVerbose="[types:8029]Bad Number\ngithub.com/pingcap/errors.AddStack\n\t/Users/sunrunaway/Code/gopath/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/Users/sunrunaway/Code/gopath/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/juju_adaptor.go:15\ngithub.com/pingcap/tidb/types.NewMaxOrMinDec\n\t/Users/sunrunaway/tidb/types/mydecimal.go:2326\ngithub.com/pingcap/tidb/types.(*testMyDecimalSuite).TestMaxOrMin\n\t/Users/sunrunaway/tidb/types/mydecimal_test.go:891\nreflect.Value.call\n\t/usr/local/Cellar/go/1.13/libexec/src/reflect/value.go:460\nreflect.Value.Call\n\t/usr/local/Cellar/go/1.13/libexec/src/reflect/value.go:321\ngithub.com/pingcap/check.(*suiteRunner).forkTest.func1\n\t/Users/sunrunaway/Code/gopath/pkg/mod/github.com/pingcap/check@v0.0.0-20191216031241-8a5a85928f12/check.go:850\ngithub.com/pingcap/check.(*suiteRunner).forkCall.func1\n\t/Users/sunrunaway/Code/gopath/pkg/mod/github.com/pingcap/check@v0.0.0-20191216031241-8a5a85928f12/check.go:739\nruntime.goexit\n\t/usr/local/Cellar/go/1.13/libexec/src/runtime/asm_amd64.s:1357"] [stack="github.com/pingcap/log.Error\n\t/Users/sunrunaway/Code/gopath/pkg/mod/github.com/pingcap/log@v0.0.0-20200117041106-d28c14d3b1cd/global.go:42\ngithub.com/pingcap/parser/terror.Log\n\t/Users/sunrunaway/Code/gopath/pkg/mod/github.com/pingcap/parser@v0.0.0-20200207090844-d65f5147dd9f/terror/terror.go:363\ngithub.com/pingcap/tidb/types.NewMaxOrMinDec\n\t/Users/sunrunaway/tidb/types/mydecimal.go:2326\ngithub.com/pingcap/tidb/types.(*testMyDecimalSuite).TestMaxOrMin\n\t/Users/sunrunaway/tidb/types/mydecimal_test.go:891\nreflect.Value.call\n\t/usr/local/Cellar/go/1.13/libexec/src/reflect/value.go:460\nreflect.Value.Call\n\t/usr/local/Cellar/go/1.13/libexec/src/reflect/value.go:321\ngithub.com/pingcap/check.(*suiteRunner).forkTest.func1\n\t/Users/sunrunaway/Code/gopath/pkg/mod/github.com/pingcap/check@v0.0.0-20191216031241-8a5a85928f12/check.go:850\ngithub.com/pingcap/check.(*suiteRunner).forkCall.func1\n\t/Users/sunrunaway/Code/gopath/pkg/mod/github.com/pingcap/check@v0.0.0-20191216031241-8a5a85928f12/check.go:739"]

----------------------------------------------------------------------
FAIL: convert_test.go:973: testTypeConvertSuite.TestConvertJSONToDecimal

convert_test.go:990:
    c.Assert(err, IsNil, Commentf("input: %v, casted: %v, out: %v, json: %#v", tt.In, casted, tt.Out, j))
... value *errors.withStack = [types:1690]%s value is out of range in '%s' ("[types:1690]%s value is out of range in '%s'")
... input: "1234567890123456789012345678901234567890123456789012345", casted: 890123456789012345, out: 1234567890123456789012345678901234567890123456789012345, json: json.BinaryJSON{TypeCode:0xc, Value:[]uint8{0x37, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35}}
```

--- a/types/mydecimal_test.go
+++ b/types/mydecimal_test.go
@@ -21,8 +21,13 @@ import (
 )
 
 var _ = Suite(&testMyDecimalSuite{})
+var _ = SerialSuites(&testMyDecimalSerialSuite{})
 
 type testMyDecimalSuite struct {
+}
+
+// testMyDecimalSerialSuite hold test cases that must run in serial
+type testMyDecimalSerialSuite struct {
 }
 
 func (s *testMyDecimalSuite) TestFromInt(c *C) {
@@ -254,7 +259,8 @@ func (s *testMyDecimalSuite) TestRemoveTrailingZeros(c *C) {
 	}
 }
 
-func (s *testMyDecimalSuite) TestShift(c *C) {
+// this test will change global variable `wordBufLen`, so it must run in serial
+func (s *testMyDecimalSerialSuite) TestShift(c *C) {
 	type tcase struct {
 		input  string
 		shift  int
@@ -472,7 +478,8 @@ func (s *testMyDecimalSuite) TestRoundWithCeil(c *C) {
 	}
 }
 
-func (s *testMyDecimalSuite) TestFromString(c *C) {
+// this test will change global variable `wordBufLen`, so it must run in serial
+func (s *testMyDecimalSerialSuite) TestFromString(c *C) {
 	type tcase struct {
 		input  string
 		output string

